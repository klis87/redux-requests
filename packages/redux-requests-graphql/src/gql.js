export const gql = (strings, ...values) =>
  strings
    .map((string, i) => string + (values[i] || ''))
    .join('')
    .trim();
