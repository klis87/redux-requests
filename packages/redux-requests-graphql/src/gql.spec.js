import { gql } from './gql';

describe('gql', () => {
  it('handles queries without variables', () => {
    expect(gql`
      query {
        x
      }
    `).toMatchSnapshot();
  });

  it('handles queries with variables', () => {
    expect(gql`
      query($id: ID!) {
        x(id: $id) {
          y
        }
      }
    `).toMatchSnapshot();
  });

  it('handles fragments', () => {
    const bookFragment = gql`
      fragment BookFragment on Book {
        id
        title
      }
    `;

    expect(gql`
      query($id: ID!) {
        book(id: $id) {
          ...BookFragment
        }
      }
      ${bookFragment}
    `).toMatchSnapshot();
  });
});
