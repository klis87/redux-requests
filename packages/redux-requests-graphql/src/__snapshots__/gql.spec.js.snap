// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gql handles fragments 1`] = `
"query($id: ID!) {
        book(id: $id) {
          ...BookFragment
        }
      }
      fragment BookFragment on Book {
        id
        title
      }"
`;

exports[`gql handles queries with variables 1`] = `
"query($id: ID!) {
        x(id: $id) {
          y
        }
      }"
`;

exports[`gql handles queries without variables 1`] = `
"query {
        x
      }"
`;
